pkg/route/routeapihelpers/routeapihelpers_test.go

```diff
@@ -299,3 +299,67 @@ func TestIngressURI(t *testing.T) {
 		})
 	}
 }
+
+// TestValidateHost ensures not specifying a proper host name results in error and
+// that a correctly specified host name passes successfully
+func TestValidateRoute(t *testing.T) {
+	tests := []struct {
+		name              string
+		host              string
+		allowNonCompliant string
+		expectedErrors    int
+	}{
+		{
+			name:              "Non-DNS-compliant host with override annotation",
+			host:              "host",
+			allowNonCompliant: "true",
+			expectedErrors:    0,
+		},
```

Comment:
Could you add another test case or two with `allowNonCompliant: "true"` to show what it does reject?